### PR TITLE
Correlated query : wrong column lookup issue while unparsing of correlated variable

### PR DESCRIPTION
**Pull request by naveenantonie**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -866,6 +871,17 @@ public SqlNode toSql(@Nullable RexProgram program, RexNode rex) {
       }
     }
 
+    private SqlNode getSqlNodeByName(Context correlAliasContext, RexFieldAccess lastAccess) {
+      SqlNode node = null;
+      for (SqlNode sqlNode : correlAliasContext.fieldList()) {
```
Comment (vinayujeedm) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:876]: We can avoid unnecessary iteration of loop.

private SqlNode getSqlNodeByName(Context correlAliasContext, RexFieldAccess lastAccess) {
    for (SqlNode sqlNode : correlAliasContext.fieldList()) {
        if (isMatchingSqlNode(sqlNode, lastAccess)) {
            return sqlNode;
        }
    }
    return null;
}

**Review by vinayujeedm — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java (hunk 1 above)
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:875]: done

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 2]
```diff
@@ -1410,6 +1426,21 @@ private SqlNode op(SqlOperator op, C value) {
     }
   }
 
+  private static boolean isNodeMatching(SqlNode node, RexFieldAccess lastAccess) {
+    boolean qualified = node.toString().split("\\.").length > 1;
+    boolean match;
+    if (qualified) {
```
[the remark this answers is not in the document]
Comment (naveenantonie) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: done

**Review by vinayujeedm — APPROVED** [review]

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**